src/ape_geth/provider.py

```diff
@@ -469,4 +471,23 @@ def build_command(self) -> List[str]:
 
 # NOTE: The default behavior of EthereumNodeBehavior assumes geth.
 class Geth(EthereumNodeProvider):
-    pass
+    @property
+    def uri(self) -> str:
+        uri = super().uri
+        ecosystem = self.network.ecosystem.name
+        network = self.network.name
+
+        # If we didn't find one in config, look for a public RPC.
+        if not uri or uri == DEFAULT_SETTINGS["uri"]:
```

Comment:
A simple test here for this new logic would be complete the story

Reply:
How's this?  8c2cf999

Reply:
Disliking that these [make actual RPC calls though](https://github.com/ApeWorX/ape/actions/runs/7588778920/job/20672029316?pr=1866).  Thoughts?

Reply:
We could do this:

```python
def test_default_public_uri(config):
    cfg = config.get_config("ethereum").sepolia
    assert cfg.uri in PUBLIC_CHAIN_RPCS["ethereum"]["sepolia"]
```

It is simpler, won't be flakey, and still tests the changes, I think, since the URI is calculated at the config level.

If you prefer the integration test though, we could catch the exception and handle it like task-failed-successfully.

Reply:
Good idea.  Done in [f8dcccd](https://github.com/ApeWorX/ape/pull/1866/commits/f8dcccd3e7bdf95adeb24ff9d6c6805b4570e5b9)